api/metrics/v1/rules.go

```diff
@@ -98,7 +138,13 @@ func (rh *rulesHandler) get(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	enforceLabelsInRules(rawRules, rh.tenantLabel, id)
+	err = enforceLabelsInRules(rawRules, rh.tenantLabel, id)
+	if err != nil {
+		level.Error(rh.logger).Log("msg", "could not enforce labels in rules", "err", err.Error())
+		http.Error(w, "error enforcing labels in rules", http.StatusInternalServerError)
```

Comment:
Nit: I feel like this might be a bit cryptic in the response for the user, maybe something along the lines 'failed to process rules' would suffice.